### ISSUE TITLE
[objects.layout] Use flexbox

### ISSUE THREAD

**Issue by florianbouvot** [inuitcss/inuitcss#380]

**Comment by herzinger** [comment]
I usually use this project: https://github.com/nicoqh/inuit-flexgrid

It's probably a good reference.

**Comment by florianbouvot** [comment]
@herzinger agree, there are more options for alignement... (with a way to configure it). I admit that I often work with a simple object and use flex box utilities to go further (because of responsive issue - modifiers are not responsive...).

**Comment by kkalvaa** [comment]
Layout should be using grid, not flexbox by now. With 88.35% global support according to Can I Use (https://caniuse.com/#search=grid) grid is now mature enough to be the preferred method for laying out content in 2 dimensions. Flex is great for 1 dimension, but comes up short in 2.

Then again I'd argue grid has removed the need for a grid system in a css framework.

**Comment by solleer** [comment]
Layout does not specify how many rows there are while the grid property requires that to be set ahead of time. They fulfill different uses.

**Comment by kkalvaa** [comment]
No, you do not. The browser will create as many implicit rows as necessary. Ref.: https://gridbyexample.com/examples/example10/

If you define a 4 column grid and put 12 children in there the browser will then automatically create 3 rows. If one of those children is set to span 4 columns the grid will now have 4 rows without you setting anything. You can also set a child to span X rows, something you cannot do with flexbox or inline-block.

You can explicitly set rows, which is nice if you have specific areas in mind, but you do not have to, the browser will do it for you.

**Comment by herzinger** [comment]
As you said, there's no need for a framework structure to use css grids. Inuitcss is fully modular, if you feel comfortable using grid for everything, simply don't import the layout object.
